pkg/cmd/roachtest/tests/failover.go

```diff
@@ -717,6 +726,52 @@ func (f *crashFailer) Recover(ctx context.Context, nodeID int) {
 	f.c.Start(ctx, f.t.L(), f.startOpts, f.startSettings, f.c.Node(nodeID))
 }
 
+// diskStallFailer stalls the disk indefinitely. This should cause the node to
+// eventually self-terminate, but we'd want leases to move off before then.
+type diskStallFailer struct {
+	t             test.Test
+	c             cluster.Cluster
+	m             cluster.Monitor
+	startOpts     option.StartOpts
+	startSettings install.ClusterSettings
+}
+
+func (f *diskStallFailer) Setup(ctx context.Context) {
+	f.c.Run(ctx, f.c.All(), `sudo umount /mnt/data1`)
+	f.c.Run(ctx, f.c.All(), `sudo dmsetup remove_all`)
+	f.c.Run(ctx, f.c.All(), `echo "0 $(sudo blockdev --getsz /dev/nvme0n1) linear /dev/nvme0n1 0" | `+
+		`sudo dmsetup create data1`)
+	f.c.Run(ctx, f.c.All(), `sudo mount /dev/mapper/data1 /mnt/data1`)
+}
+
+func (f *diskStallFailer) Ready(ctx context.Context, m cluster.Monitor) {
+	f.m = m
+}
+
+func (f *diskStallFailer) Cleanup(ctx context.Context) {
+	f.c.Run(ctx, f.c.All(), `sudo dmsetup resume data1`)
+	// We have to stop the cluster to remount /mnt/data1.
+	f.m.ExpectDeaths(int32(f.c.Spec().NodeCount))
```

Comment:
You're probably getting away with it now, but a test that has a dedicated workload node will have trouble here because you'll be killing the workload too (?) but also expecting one more death than you're seeing (which I think can mask later crashes).

You could consider letting the failer operate on a subset of nodes, or just add a comment for now that whoever needs this in the future (if it ever happens) needs to make some changes.

Reply:
This happens during the cleanup stage after the test is done (either successfully or failed). At this point we don't care what happens to the cluster or workload.